### PR TITLE
feat: Installation script optimizations

### LINKED ISSUE #10660: Installation script improvements
Ticket to address technical debt on installation script

### PR DESCRIPTION
**Pull request by hackergil**
Fixes #10660

### CODE REVIEW

File: scripts/install/run.sh [hunk 1]
```diff
@@ -299,54 +289,55 @@ function stop_apps {
...
+if [ -z "${1}" ]; then
     cmd_help
     exit 1
 fi
 
-readonly commands="$1"
+readonly commands="${1}"
 
-echo "Loading configs from ./config.default.sh"
-    . ./config.default.sh
```
Comment (mirkosoft) [inline scripts/install/run.sh:329]: this change caused the default configuration is not loaded (only `config.sh` file is being loaded)

File: scripts/install/run.sh [hunk 2]
```diff
@@ -215,74 +205,74 @@ function local_install {
     echo "Finished: npm @spartacus:registry set back to https://registry.npmjs.org/"
 }
 
-function prestart_csr {
-    if [ -z "$CSR_PORT" ]; then
-        echo "Skipping prestart csr script"
+function build_csr {
+    if [ -z "${CSR_PORT}" ]; then
+        echo "Skipping csr app build"
```
Comment (mirkosoft) [inline scripts/install/run.sh]: let's add ` (no port defined)` here as well

File: scripts/install/run.sh [hunk 3]
```diff
@@ -215,74 +205,74 @@ function local_install {
...
+        ( cd ${INSTALLATION_DIR}/spartacus && yarn build --prod )
     fi
 }
 
-function prestart_ssr {
-    if [ -z "$SSR_PORT" ]; then
-        echo "Skipping prestart ssr script"
+function build_ssr {
+    if [ -z "${SSR_PORT}" ]; then
+        echo "Skipping ssr app build"
```
Comment (mirkosoft) [inline scripts/install/run.sh]: let's add ` (no port defined)`

File: scripts/install/run.sh [hunk 4]
```diff
@@ -215,74 +205,74 @@ function local_install {
...
+        ( cd ${INSTALLATION_DIR}/spartacus-ssr && yarn build && yarn build:ssr )
     fi
 }
 
-function prestart_ssr_pwa {
-    if [ -z "$SSR_PWA_PORT" ]; then
-        echo "Skipping prestart ssr script (with pwa support)"
+function build_ssr_pwa {
+    if [ -z "${SSR_PWA_PORT}" ]; then
+        echo "Skipping ssr app build (with pwa support)"
```
Comment (mirkosoft) [inline scripts/install/run.sh]: let's add ` (no port defined)`

File: scripts/install/run.sh [hunk 5]
```diff
@@ -215,74 +205,74 @@ function local_install {
...
-        ( cd ${INSTALLATION_DIR}/ssr-pwa && yarn build && yarn build:ssr )
+        printh "Building ssr app (with pwa support)"
+        ( cd ${INSTALLATION_DIR}/spartacus-ssr-pwa && yarn build && yarn build:ssr )
     fi
 }
 
 function start_csr_unix {
-    if [ -z "$CSR_PORT" ]; then
+    if [ -z "${CSR_PORT}" ]; then
         echo "Skipping csr app start"
```
Comment (mirkosoft) [inline scripts/install/run.sh]: let's add ` (no port defined)`

File: scripts/install/run.sh [hunk 6]
```diff
@@ -215,74 +205,74 @@ function local_install {
...
         printh "Starting csr app"
-        pm2 start --name "csr-${CSR_PORT}" serve -- ${INSTALLATION_DIR}/csr/dist/csr/ --single -p ${CSR_PORT}
+        pm2 start --name "csr-${CSR_PORT}" serve -- ${INSTALLATION_DIR}/spartacus/dist/spartacus/ --single -p ${CSR_PORT}
     fi
 }
 
 function start_ssr_unix {
-     if [ -z "$SSR_PORT" ]; then
+     if [ -z "${SSR_PORT}" ]; then
         echo "Skipping ssr app start"
```
Comment (mirkosoft) [inline scripts/install/run.sh]: let's add ` (no port defined)`

File: scripts/install/run.sh [hunk 7]
```diff
@@ -215,74 +205,74 @@ function local_install {
...
         printh "Starting ssr app"
-        ( cd ${INSTALLATION_DIR}/ssr && export PORT=${SSR_PORT} && export NODE_TLS_REJECT_UNAUTHORIZED=0 && pm2 start --name "ssr-${SSR_PORT}" dist/ssr/server/main.js )
+        ( cd ${INSTALLATION_DIR}/spartacus-ssr && export PORT=${SSR_PORT} && export NODE_TLS_REJECT_UNAUTHORIZED=0 && pm2 start --name "ssr-${SSR_PORT}" dist/spartacus-ssr/server/main.js )
     fi
 }
 
 function start_ssr_pwa_unix {
-     if [ -z "$SSR_PWA_PORT" ]; then
+     if [ -z "${SSR_PWA_PORT}" ]; then
         echo "Skipping ssr (with pwa support) app start"
```
Comment (mirkosoft) [inline scripts/install/run.sh]: let's add ` (no port defined)`

File: scripts/install/run.sh [hunk 8]
```diff
@@ -215,74 +205,74 @@ function local_install {
...
-    if [ -z "$CSR_PORT" ]; then
-        echo "Skipping prestart csr script"
+function build_csr {
+    if [ -z "${CSR_PORT}" ]; then
+        echo "Skipping csr app build"
     else
-        printh "Prestart setup for csr app"
-        ( cd ${INSTALLATION_DIR}/csr && yarn build --prod )
+        printh "Building csr app"
+        ( cd ${INSTALLATION_DIR}/spartacus && yarn build --prod )
```
Comment (mirkosoft) [inline scripts/install/run.sh]: Can we get rid of this hardcoded app name and use param instead?

File: scripts/install/run.sh [hunk 9]
```diff
@@ -215,74 +205,74 @@ function local_install {
...
-    if [ -z "$SSR_PORT" ]; then
-        echo "Skipping prestart ssr script"
+function build_ssr {
+    if [ -z "${SSR_PORT}" ]; then
+        echo "Skipping ssr app build"
     else
-        printh "Prestart setup for ssr app"
-        ( cd ${INSTALLATION_DIR}/ssr && yarn build && yarn build:ssr )
+        printh "Building ssr app"
+        ( cd ${INSTALLATION_DIR}/spartacus-ssr && yarn build && yarn build:ssr )
```
Comment (mirkosoft) [inline scripts/install/run.sh]: Can we get rid of this hardcoded app name and use param instead?

File: scripts/install/run.sh [hunk 10]
```diff
@@ -215,74 +205,74 @@ function local_install {
...
-    if [ -z "$SSR_PWA_PORT" ]; then
-        echo "Skipping prestart ssr script (with pwa support)"
+function build_ssr_pwa {
+    if [ -z "${SSR_PWA_PORT}" ]; then
+        echo "Skipping ssr app build (with pwa support)"
     else
-        printh "Prestart setup for ssr app (with pwa support)"
-        ( cd ${INSTALLATION_DIR}/ssr-pwa && yarn build && yarn build:ssr )
+        printh "Building ssr app (with pwa support)"
+        ( cd ${INSTALLATION_DIR}/spartacus-ssr-pwa && yarn build && yarn build:ssr )
```
Comment (mirkosoft) [inline scripts/install/run.sh]: Can we get rid of this hardcoded app name and use param instead?

File: scripts/install/run.sh [hunk 11]
```diff
@@ -215,74 +205,74 @@ function local_install {
...
 function start_csr_unix {
-    if [ -z "$CSR_PORT" ]; then
+    if [ -z "${CSR_PORT}" ]; then
         echo "Skipping csr app start"
     else
-        prestart_csr
+        build_csr
         printh "Starting csr app"
-        pm2 start --name "csr-${CSR_PORT}" serve -- ${INSTALLATION_DIR}/csr/dist/csr/ --single -p ${CSR_PORT}
+        pm2 start --name "csr-${CSR_PORT}" serve -- ${INSTALLATION_DIR}/spartacus/dist/spartacus/ --single -p ${CSR_PORT}
```
Comment (mirkosoft) [inline scripts/install/run.sh]: Can we get rid of this hardcoded app name and use param instead?

File: scripts/install/run.sh [hunk 12]
```diff
@@ -215,74 +205,74 @@ function local_install {
...
 function start_ssr_unix {
-     if [ -z "$SSR_PORT" ]; then
+     if [ -z "${SSR_PORT}" ]; then
         echo "Skipping ssr app start"
     else
-        prestart_ssr
+        build_ssr
         printh "Starting ssr app"
-        ( cd ${INSTALLATION_DIR}/ssr && export PORT=${SSR_PORT} && export NODE_TLS_REJECT_UNAUTHORIZED=0 && pm2 start --name "ssr-${SSR_PORT}" dist/ssr/server/main.js )
+        ( cd ${INSTALLATION_DIR}/spartacus-ssr && export PORT=${SSR_PORT} && export NODE_TLS_REJECT_UNAUTHORIZED=0 && pm2 start --name "ssr-${SSR_PORT}" dist/spartacus-ssr/server/main.js )
```
Comment (mirkosoft) [inline scripts/install/run.sh]: Can we get rid of this hardcoded app name and use param instead?

File: scripts/install/run.sh [hunk 13]
```diff
@@ -215,74 +205,74 @@ function local_install {
...
 function start_ssr_pwa_unix {
-     if [ -z "$SSR_PWA_PORT" ]; then
+     if [ -z "${SSR_PWA_PORT}" ]; then
         echo "Skipping ssr (with pwa support) app start"
     else
-        prestart_ssr_pwa
+        build_ssr_pwa
         printh "Starting ssr app (with pwa support)"
-        ( cd ${INSTALLATION_DIR}/ssr-pwa && export PORT=${SSR_PWA_PORT} && export NODE_TLS_REJECT_UNAUTHORIZED=0 && pm2 start --name "ssr-pwa-${SSR_PWA_PORT}" dist/ssr/server/main.js )
+        ( cd ${INSTALLATION_DIR}/ssr-pwa && export PORT=${SSR_PWA_PORT} && export NODE_TLS_REJECT_UNAUTHORIZED=0 && pm2 start --name "ssr-pwa-${SSR_PWA_PORT}" dist/spartacus-ssr-pwa/server/main.js )
```
Comment (mirkosoft) [inline scripts/install/run.sh]: Can we get rid of this hardcoded app name and use param instead?

**Review by mirkosoft — CHANGES_REQUESTED** [review]
Nice clean up. I like the idea of parameters for app name for the `add_spartacus_csr` function and it's brothers. Why not to go further and create new global config vars with app names and use them everywhere in the script? See my comments on hardcoded app names.
Also `README.md` should be updated in terms of app names.

File: scripts/install/run.sh (hunk 8 above)
Comment (hackergil) [reply scripts/install/run.sh]: We are not passing any parameters to this function. Can you clarify?

File: scripts/install/run.sh (hunk 9 above)
Comment (hackergil) [reply scripts/install/run.sh]: Same as above

**Review by mirkosoft — APPROVED** [review]
All works fine.

**Pull request merged**